File: desafio-proway-2/FrontEnd/Training/src/app/models/RestRoomPerson.ts
import { Person } from './Person';
import { RestRoom } from './RestRoom';
import { RestTime } from "./RestTime";

export class RestRoomPerson {
  
  constructor() {
    this.id = 0;
    this.restTimeId = 0;
    this.restTime = new RestTime();
    this.restRoomId = 0;
    this.restRoom = new RestRoom();
    this.personId = 0;
    this.person = new Person();
  }
  
  id: number;
  restTimeId: number;
  restTime: RestTime;
  restRoomId: number;
  restRoom: RestRoom;
  personId: number;
  person: Person;
}

File: desafio-proway-2/BackEnd/Training_API/Models/RestRoomPerson.cs
namespace Training_API.Models
{
    public class RestRoomPerson
    {
        public RestRoomPerson(int id, int restTimeId, int restRoomId, int personId)
        {
            this.Id = id;
            this.RestTimeId = restTimeId;
            this.RestRoomId = restRoomId;
            this.PersonId = personId;
        }
        public int Id { get; set; }
        public int RestTimeId { get; set; }
        public RestTime RestTime { get; set; }
        public int RestRoomId { get; set; }
        public RestRoom RestRoom { get; set; }
        public int PersonId { get; set; }
        public Person Person { get; set; }
    }
}